markdown/extensions/fenced_code.py

```diff
@@ -66,29 +69,41 @@ def run(self, lines):
         while 1:
             m = self.FENCED_BLOCK_RE.search(text)
             if m:
-                lang = ''
-                if m.group('lang'):
-                    lang = self.LANG_TAG % m.group('lang')
+                lang = None
+                id, classes, config = '', [], {}
+                if m.group('attrs'):
+                    id, classes, config = self.handle_attrs(get_attrs(m.group('attrs')))
+                    if len(classes):
+                        lang = classes[0]
+                elif m.group('lang'):
+                    lang = m.group('lang')
+                    classes.append(lang)
 
                 # If config is not empty, then the codehighlite extension
                 # is enabled, so we call it to highlight the code
                 if self.codehilite_conf:
+                    local_config = self.codehilite_conf.copy()
+                    local_config.update(config)
+                    # TODO: pass id and classes to codehilite
```

Comment:
Thanks for this work. My use case is to put a class on the outer div that codehilite generates. In that case you can just append `' '.join(classes)` to the css_class parameter passed to CodeHilite. It looks like it will take some changes to CodeHilite itself to allow passing an id in.